TF_Tooling_Web/src/Plugins/InputActionManager.ts

```diff
@@ -0,0 +1,73 @@
+import { TouchFreeInputAction } from "../TouchFreeToolingTypes";
+import { InputActionPlugin } from "./InputActionPlugin";
+
+// Class: InputActionManager
+// The manager for all <TouchFreeInputActions> to be handled and distributed. This runs the
+// received data through any <InputActionPlugins> given to it and finaly distributes the data
+// via the  <TransmitInputAction> event which should be listened to by any class hoping to make
+// use of incoming <TouchFreeInputActions>.
+export class InputActionManager extends EventTarget {
+    // Event: TransmitInputAction
+    // An event for transmitting <TouchFreeInputActions> that are received via the <messageReceiver> to
+    // be listened to.
+
+    // Event: TransmitInputActionRaw
+    // An event for immediately transmitting <TouchFreeInputActions> that are received via the
+    // <messageReceiver>to be listened to.
+
+    // Variable: instance
+    // The instance of the singleton for referencing the events transmitted
+    static _instance: InputActionManager;
+
+    static plugins: Array<InputActionPlugin>;
+
+    public static get instance() {
+        if (InputActionManager._instance == null) {
```

Comment:
I'm not entirely sure what will happen with this in JavaScript, I think that uninitialised objects will compare equal to null, but the strictly "safer" way to do this would be using `===` and in that case we might need to set it to null up on line 20.